src/botPage/bot/TradeEngine/index.js

```diff
@@ -17,18 +17,30 @@ import rootReducer from './state/reducers';
 import * as constants from './state/constants';
 import { start } from './state/actions';
 
+let prevTickValue = 0;
 const watchBefore = store =>
     watchScope({
         store,
-        stopScope: constants.DURING_PURCHASE,
-        passScope: constants.BEFORE_PURCHASE,
-        passFlag : 'proposalsReady',
+        stopScope    : constants.DURING_PURCHASE,
+        passScope    : constants.BEFORE_PURCHASE,
+        passCondition: state => {
+            if (state.proposalsReady && (!prevTickValue || state.newTick !== prevTickValue)) {
+                prevTickValue = state.newTick;
+                return true;
+            }
+            return false;
+        },
     });
 
 const watchDuring = store =>
-    watchScope({ store, stopScope: constants.STOP, passScope: constants.DURING_PURCHASE, passFlag: 'openContract' });
+    watchScope({
+        store,
+        stopScope    : constants.STOP,
+        passScope    : constants.DURING_PURCHASE,
+        passCondition: state => state.openContract,
```

Comment:
It seems like the problem can still happen in here. As both `proposal_open_contract` and `ticks` can trigger the condition.

A more general purpose solution would be to move the `lastTick` check to the `watchScope` function.

For simplicity `watchScope` can return a closure containing the `lastTick` variable.

```
const watchScope = ({ store, stopScope, passScope, passCondition }) => {
  ...
  let lastTick;
  return new Promise(resolve => {
    const unsubscribe = store.subscribe(() => {
      if (newState.newTick === lastTick) return;
      lastTick = newState.newTick;

      ... // The rest can be the same as before, as we don't check the ticks anymore we can used the flag instead of the condition
        
}
```

Reply:
Done

Reply:
I don't think we need `passCondition` anymore, better to move it back to `passFlag`.